proxy/healthy_endpoints_test.go

```diff
@@ -68,15 +71,28 @@ func setupProxy(t *testing.T, doc string) (*metricstest.MockMetrics, *httptest.S
 	return m, ps
 }
 
-func TestPHCWithoutRequests(t *testing.T) {
+func setupServices(t *testing.T, healthy, unhealthy int) []*httptest.Server {
 	services := []*httptest.Server{}
-	for i := 0; i < 3; i++ {
+	for i := 0; i < healthy; i++ {
 		service := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			w.WriteHeader(http.StatusOK)
 		}))
 		services = append(services, service)
-		defer service.Close()
+		t.Cleanup(service.Close)
 	}
+	for i := 0; i < unhealthy; i++ {
+		service := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			time.Sleep(100 * time.Millisecond)
+			w.WriteHeader(http.StatusOK)
+		}))
+		services = append(services, service)
+		t.Cleanup(service.Close)
+	}
+	return services
```

Comment:
Since you are doing automatic cleanup, it might be handy to return slice of service URLs if its the only thing that is used by the caller.